SwiftRuntimeLibrary/BaseAssociatedTypeProxy.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using SwiftRuntimeLibrary.SwiftMarshal;
+
+namespace SwiftRuntimeLibrary {
+	namespace ClassWrapTests {
+		[SwiftNativeObject ()]
+		public class BaseAssociatedTypeProxy : ISwiftObject {
+			IntPtr handle;
+			SwiftMetatype class_handle;
+			SwiftObjectFlags object_flags = SwiftObjectFlags.IsSwift;
+
+			protected BaseAssociatedTypeProxy (IntPtr handle, SwiftMetatype classHandle, SwiftObjectRegistry registry)
+			{
+				if (SwiftNativeObjectAttribute.IsSwiftNativeObject (this)) {
+					object_flags |= SwiftObjectFlags.IsDirectBinding;
+				}
+				class_handle = classHandle;
+				SwiftObject = handle;
+				if (IsCSObjectProxy)
+					registry.Add (this);
+			}
+
+			protected BaseAssociatedTypeProxy (byte [] swiftTypeData, SwiftMetatype mt)
+			{
+				if (swiftTypeData == null)
+					throw new ArgumentNullException (nameof (swiftTypeData));
+				var length = swiftTypeData.Length;
+				StructMarshal.Marshaler.RetainNominalData (mt, swiftTypeData);
+				SwiftData = new byte [length];
+				Array.Copy (swiftTypeData, SwiftData, length);
```

Comment:
I generally prefer `Buffer.BlockCopy` for `byte` arrays... but I'm not sure in 2020 who's the fastest so.

Reply:
Probably the fastest for large blocks is to pinvoke into `memcpy` (see [here](https://stackoverflow.com/a/1445405/20481) for some pinvoke chicanery I did years ago for `memcmp`, nearly the same task). Considering that the data is likely to be << 1K, I doubt that there will be much difference.